starter/source/coupling/rad2rad/r2r_group.F

```diff
@@ -417,17 +419,42 @@ SUBROUTINE R2R_GROUP(
 C---------------FLAG = 1 --> Creation of groups and links------------C
 C--------------------------------------------------------------------C
           INNOD = 0
-          DO K=1,4
+          NN = 4
+C
+C---------------Tag of nodes with nlocal dof-------------------------C
+          IF (NLOC_DMG%IMOD > 0) THEN
+            NN = 5
+            MY_ALLOCATE(TAG_NLOCAL,NUMNOD)
+            TAG_NLOCAL(1:NUMNOD) = 0
+            CALL TAGNOD_R2R_NL(IXC,IXTG,IXS,IXS10,IXS20,
+     .                         IXS16,TAG_NLOCAL,IPM)
+          ENDIF
+C
+          DO K=1,NN
             COMPT = 0
             IAD_TMP = IAD
-C---------------Storage inside buffer--------------------------------C	
-            DO J=1,NUMNOD
-              IF (TAGNO(J+NPART)==(K+N)) THEN 
-                BUF_NOD(IAD)=J
-                IAD=IAD+1
-                COMPT = COMPT+1	        
-              ENDIF
-            ENDDO
+C-----------Storage inside buffer------------------------------------C
+            IF (K < 5) THEN
+              DO J=1,NUMNOD	     
+                IF (TAGNO(J+NPART)==(K+N)) THEN 
+                  BUF_NOD(IAD)=J
+                  IAD=IAD+1
+                  COMPT = COMPT+1	        
+                ENDIF
+              ENDDO
+            ELSE
+C-----------Nodes in link type 4 + nlocal dof -> aditional link------C
+C-----------Node coupled only if nl material on both sides ----------C 
+              DO J=1,NUMNOD
+                IF ((TAG_NLOCAL(J)==1).AND.(TAGNO(J+NPART+NUMNOD) == N+1)) THEN
+                  BUF_NOD(IAD)=J
+                  IAD=IAD+1
+                  COMPT = COMPT+1	        
+                ENDIF
+              ENDDO
+              DEALLOCATE(TAG_NLOCAL)
```

Comment:
@bertrand-altair 

I'm not sure dealloc it at good position here  : if K < 5 , it is never deallocated

Reply:
Not a bug as when array is allocated K=5 is always reached, but i have clean the code anyway.